pages/recipes/[id]/edit.js

```diff
@@ -0,0 +1,41 @@
+import { useRouter } from "next/router";
+import useSWR from "swr";
+import { StyledLink } from "@/components/StyledLink";
+import RecipeForm from "@/components/Formular/FormularAddRecipe";
+
+export default function EditPage() {
+  const router = useRouter();
+  const { id } = router.query;
+  const { data: recipe, error } = useSWR(`/api/recipes/${id}`);
```

Comment:
I recommend changing this line of code to contain a ternary operator:

```suggestion
const { data: recipe, isLoading, error, } = useSWR(id ? `/api/recipes/${id}` : null);
```

Without the check, you get an error on your server
`An error occurred: CastError: Cast to ObjectId failed for value "undefined" (type string) at path "_id" for model "Recipe"`
Since at the very start, the `id` you sent to your api route is undefined